audio-worklet/src/noisegate-demo.js

```diff
@@ -0,0 +1,284 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+class NoisegateDemo {
+  /**
+   * @class NoisegateDemo
+   * @constructor Initializes and modifies noisegate settings in response to
+   * GUI input.
+   * @param {AudioContext} context The Audio Context
+   * @param {Boolean} workletIsAvailable Switch depending on browser
+   *                  which determines if both script processor and audio
+   *                  worklet should be used. If false, only the script
+   *                  processor will be used.
+   */
+  constructor(context, workletIsAvailable) {
+    this.context_ = context;
+    this.masterGain_ = new GainNode(this.context_, {gain: 0.5});
+    this.workletIsAvailable_ = workletIsAvailable;
+
+    this.attack_ = 0;
+    this.attackMax_ = 1;
+    this.attackMin_ = 0;
+    this.release_ = 0;
+    this.releaseMin_ = 0;
+    this.releaseMax_ = 1;
+    this.threshold_ = -100;
+    this.thresholdMin_ = -100;
+    this.thresholdMax_ = 0;
+
+    // TODO: Implement audio worklet logic.
+
+    const scriptProcessorBufferSize = 4096;
+    this.noisegateScriptProcessor_ = new NoiseGate(this.context_, {
+        channelCount: 1,
+        attack: this.attack_,
+        release: this.release_,
+        threshold: this.threshold_
+    });
+
+    // There are two sound sources (speech and noise), which are mixed
+    // together. The user can set the gain of the noise, and also whether or
+    // not the mixed sound is processed by the noise gate.
+    this.speechGain_ = new GainNode(this.context_);
+    this.noiseGain_ = new GainNode(this.context_);
+    this.speechAndNoiseGain_ = new GainNode(this.context_);
+    this.withoutNoisegateGain_ = new GainNode(this.context_, {gain: 0});
+    this.noisegateGain_ = new GainNode(this.context_);
+    this.scriptProcessorGain_ = new GainNode(this.context_);
+
+    this.noiseGain_.connect(this.speechAndNoiseGain_);
+    this.speechGain_.connect(this.speechAndNoiseGain_);
+    this.speechAndNoiseGain_.connect(this.withoutNoisegateGain_)
+        .connect(this.masterGain_);
+    
+    this.speechAndNoiseGain_.connect(this.noisegateScriptProcessor_.input);
+    this.noisegateScriptProcessor_.output.connect(this.scriptProcessorGain_)
+        .connect(this.noisegateGain_).connect(this.masterGain_);
+
+    this.masterGain_.connect(this.context_.destination);
+
+    this.loadSound('sound/speech.ogg').then((sound) => {
```

Comment:
I agree. I made it a bit longer and created a custom audio recording with no pauses within the sentence.